src/main/java/seedu/address/model/person/Id.java

```diff
@@ -0,0 +1,56 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Person's ID in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidId(String)}
+ */
+public class Id {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "ID should only contain alphanumeric characters, and it should not be blank";
+    public static final String VALIDATION_REGEX = "^[A-Z0-9]+$";
```

Comment:
Consider adding support for lowercase ID types through automatic conversion to uppercase. It could offer a smoother user experience.